### DISCUSSION TITLE
just in time mode with eleventy

### DISCUSSION THREAD

**Discussion by just-website** [tailwindlabs/tailwindcss#4559] [Help]
I have a specific problem. there is a project with eleventy. I use liquid as a template engine, and i use webpack for build styles from scss. The problem appears when a lot of nested includes in template is used. At some point I losed styles in dev mode. Around half of the styles on the project are missing. It seems to me that the jit compiler works before all templates are prepared.
I want to make some kind of delay for the jit compiler to wait for build templates.
I understand that the question to a greater extent about the build config of the project, not the tailwind. I will be happy to any decisions.

**Comment by VicGUTT** [comment]
Hey !

I'm not sure I understand you issue correctly and I don't know anything about Eleventy.
But, if in your `tailwind.config.js` file you've correctly setup the `purge` array, you should be good.

```js
// tailwind.config.js
module.exports = {
    purge: [
        '/path/to/my/liquid/template.extension',
        '/path/to/all/templates/**/*.extension',
        '/path/to/all/templates2/**/*.{extension1,extension2,extension3}',
    ],
};
```

As far as I understand it, when in JIT mode, Tailwind will simply look for the specified files in your system and see which classes are used. It really doesn't do more. So if your templates exists on your system as static files and contain Tailwind classes, it should all work perfectly.

Hope I understood your issue correctly. ✌🏾

**Reply by just-website** [reply]
No, i'm talking not about that.
Eleventy is a template-based static site generator (in short). From liquid files in my case. In dev mode the templates are compiled from the watcher. I useded scripts for eleventy and building styles from scss (webpack running) in parallel.
*after my guess*:
webpack rebuild the styles. Eleventy jit compyler is triggered on new styles and tries to optimize them based on html. But the HTML has not yet been fully rebuilt. then the jit compiler removes the unused one (half of all), but it's needed.It probably because he didn't see the css classes in the html

**Reply by just-website** [reply]
everything works without config mode: 'jit'. But in my case with same purge config:
1. built main.css takes 3x weight (without jit: 360kb and 14k rows in file; with jit 112kb and 3.5k rows in file) without minify
2. In dev mode main css weight 12mb and took 160k rows in file. it hinders performance

**Reply by VicGUTT** [reply]
Hmm... ok, as I said, I really don't know much about Eleventy _(I quickly checked the docs but it was confusing so I bailed)_, nor liquid templates for that matter.
Maybe showing code samples or better yet a small reproduction demo would help.

But, just to make sure you are not in the following cases that could indeed cause Tailwind to not find used classes:
- Are you dynamically constructing the classes ? Someting like `<div class="bg-{{ $color }}-500">` ?
- Did you further customize Tailwind's purge config ?
